core/include/optee_msg.h

```diff
@@ -68,11 +68,24 @@
 #define OPTEE_MSG_ATTR_META			BIT(8)
 
 /*
- * The temporary shared memory object is not physically contiguous and this
- * temp memref is followed by another fragment until the last temp memref
- * that doesn't have this bit set.
+ * Pointer to a list of pages used to registed user-defined SHM buffer.
+ * Used with OPTEE_MSG_ATTR_TYPE_TMEM_*.
+ * buf_ptr should point to the beginning of the buffer. Buffer will contain
+ * list of page addresses. OP-TEE core can reconstruct contigous buffer from
+ * that page addresses list. Page addresses are stored as 64 bit values.
+ * Last entry on a page should point to the next page of buffer.
+ * 12 least significant of buf_ptr should hold page offset of user buffer.
+ *
+ * So, entries should be placed like members of this structure:
+ *
+ * struct page_data {
+ *	uint64_t pages_array[SMALL_PAGE_SIZE/sizeof(uint64_t) - 1];
+ *      uint64_t next_page_data;
+ * };
+ *
+ * As you can can see, it exactly fits into 4k page size.
```

Comment:
AFAIK, 4k is the smallest page size on ARM. So, this ABI will support any other page size, but in ineffective way. 

Problem is that:
1) OP-TEE can have own page size
2) Hypervisor can have own page size
3) Guest can have own page size

If OP-TEE uses 16KB page and REE uses 4KB page size, then OP-TEE will not be able to map pages from REE. If OP-TEE uses 4KB and REE uses 16KB page size, then REE will need to divide own page to smaller ones, no problem there.

I think, that easiest way is to support smallest possible page size in protocol. 

I can add `OPTEE_MSG_NONCONTIG_PAGE_SIZE`, this is not a problem.